### PR TITLE
old enum names string array support added

### PR DESCRIPTION
**Pull request by im-shiv**

### CODE REVIEW

File: bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/models/form/OptionsConstraint.java [hunk 1]
```diff
@@ -69,6 +69,16 @@ default boolean isEnforceEnum() {
      * @return the list of enum names
      * @since com.adobe.cq.forms.core.components.models.form 5.4.0
      */
-    @JsonProperty("enumNames")
+    @JsonIgnore
     TextContent[] getEnumNamesAsTextContent();
```
Comment (rismehta) [inline bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/models/form/OptionsConstraint.java:73]: Can you remove this function itself, I don't think anyone would be using it yet. Let's have one public facing API for this.

**Review by rismehta — CHANGES_REQUESTED** [review]
One comment, please check

File: bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/models/form/OptionsConstraint.java (hunk 1 above)
Comment (im-shiv) [reply bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/models/form/OptionsConstraint.java:73]: we are using it in sightly to enable rich text, eg: https://github.com/adobe/aem-core-forms-components/blob/master/ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/checkboxgroup/v1/checkboxgroup/widget.html#L28

Comment (rismehta) [reply bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/models/form/OptionsConstraint.java:73]: Please use the new API, it does not make sense to have two API's for the same thing.

Comment (im-shiv) [reply bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/models/form/OptionsConstraint.java]: got it, removed old APi

File: bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/models/form/OptionsConstraint.java [hunk 2]
```diff
@@ -63,12 +63,12 @@ default boolean isEnforceEnum() {
...
+     * Returns an array of String or TextContent to be displayed to the end user based on the richText flag.
      * The length of enum and enumNames array must match
      *
      * @return the list of enum names
-     * @since com.adobe.cq.forms.core.components.models.form 5.4.0
+     * @since com.adobe.cq.forms.core.components.models.form 5.5.0
      */
     @JsonProperty("enumNames")
-    TextContent[] getEnumNamesAsTextContent();
+    Object[] getEnumNamesArray();
```
Comment (rismehta) [inline bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/models/form/OptionsConstraint.java:73]: Let's keep this array of string, remove the `TextContent` completely from the java code. UI could still have the same flow, let's change the spec too and fix this.

**Review by rismehta — CHANGES_REQUESTED** [review]
check comments

**Pull request closed without merging**